### PR TITLE
Add migration guide from 2.x to 3.x

### PR DESCRIPTION
**Pull request by bobiechen-twilio**
Migration guide from 2.x to 3.x
- `TVOCallInvite` Changes
- Specifying a media region
- `TVOConnectOptions` & `TVOAcceptOptions`
- Ringing
- Audio Device
- Custom Parameters

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -224,6 +225,166 @@ There are number of techniques you can use to ensure that access token expiry is
...
+    <Client>
+      <Identity>bob</Identity>
+      <Parameter name="caller_first_name" value="alice" />
+      <Parameter name="caller_last_name" value="smith" />
+    </Client>
+  </Dial>
+</Response>
+```
+
+`callInvite.customParameters`:
```
Comment (kbagchiGWC) [inline README.md]: Add some text here - android says `callInvite.getCustomParameters() returns a map of key-value pair passed in the TwiML.`

**Review by kbagchiGWC — APPROVED** [review]

File: README.md [hunk 2]
```diff
@@ -224,6 +225,166 @@ There are number of techniques you can use to ensure that access token expiry is
 - Retain the access token along with the timestamp of when it was requested so you can verify ahead of time whether the token has already expired based on the `time-to-live` being used by your server.
 - Prefetch the access token whenever the `UIApplication`, or `UIViewController` associated with an outgoing call is created.
 
+## Migragion Guide
```
Comment (idelgado) [inline README.md]: Migration

File: README.md [hunk 3]
```diff
@@ -224,6 +225,166 @@ There are number of techniques you can use to ensure that access token expiry is
...
+
+// TVOCallDelegate.h
+@protocol TVOCallDelegate
+
+@optional
+- (void)callDidStartRinging:(nonnull TVOCall *)call;
+
+@end
+```
+
```
Comment (idelgado) [inline README.md:315]: Please add a section for https://github.com/twilio/voice-quickstart-android/tree/3.x#media-establishment--connectivity

**Review by idelgado — CHANGES_REQUESTED** [review]

**Review by idelgado — APPROVED** [review]

File: README.md [hunk 4]
```diff
@@ -224,6 +225,272 @@ There are number of techniques you can use to ensure that access token expiry is
...
+    if (![session overrideOutputAudioPort:AVAudioSessionPortOverrideSpeaker error:&error]) {
+        NSLog(@"AVAudiosession overrideOutputAudioPort %@",error);
+    }
+};
+audioDevice.block();
+```
+
+#### CallKit
+The Voice iOS 3.X SDK deprecates the `CallKitIntegration` category from `TwilioVoice` in favor of a new property called `TVODefaultAudioDevice.enabled`. This property provides developers with a mechanism to enable or disable the activation of the audio device prior to connecting to a Call or to stop or start the audio device while you are already connected to a Call. A Cal can now be connected without activating the audio device by setting `TVODefaultAudioDevice.enabled` to `NO` and can be enabled during the lifecycle of the Call by setting `TVODefaultAudioDevice.enabled` to `YES`. The default value is `YES`. This API change was made to ensure full compatibility with CallKit as well as supporting other use cases where developers may need to disable the audio device during a call.
```
Comment (piyushtank) [inline README.md]: sentense has a typo
`s/A Cal can/A Call can`

**Review by piyushtank — APPROVED** [review]

**Pull request merged**